Core/clim-core/presentations/presentation-type-functions.lisp

```diff
@@ -82,11 +82,9 @@ otherwise return false."
 (define-presentation-generic-function %presentation-typep presentation-typep
   (type-key parameters object type))
 
-;;; FIXME this method should return NIL. There is a problem with
-;;; MENU-ITEM translators which doesn't work without this kludge.
 (define-default-presentation-method presentation-typep (object type)
-  (declare (ignore object type))
-  t)
+  (declare (ignore object))
+  (warn "Presentation type ~s doesn't implement PRESENTATION-TYPEP." type))
```

Comment:
I don't think warnings work very well for this purpose (see "Drawing outside the mirror" warning). If not having a method is a violation of the protocol, this should be an error.

Reply:
it is not a violation, but it is a likely issue in the user code to call presentation-typep on a presentation without this method

Reply:
(at least it was a likely issue in McCLIM code :-)

Reply:
I guess the questions are "What do I have to do when I see this warning for one of my presentation types?" and "What happens if I don't do anything?".

Reply:
presentation-typep is not called on user-defined presentation types from mcclim core, so you, as a caller of the presentation-typep, know best what you need to do (i.e replace the call with hardcoded nil, or implement the method).

Reply:
Resolved in IRC discussion.